### PR TITLE
Include a native module preparation guide …

### PR DESCRIPTION
**Pull request by jugutier**
with code scaffolding tool so that readers can jump straight into the code

Thank you for the PR! Contributors like you keep React Native awesome!

Please see the Contribution Guide for guidelines:

https://github.com/facebook/react-native-website/blob/master/CONTRIBUTING.md

If your PR references an existing issue, please add the issue number below:

#163

### CODE REVIEW

**Comment by jugutier** [comment]
sidebar.json doesn't seem to adding my new section in this preview: https://deploy-preview-169--react-native.netlify.com

Did I miss something?

**Comment by jugutier** [comment]
thanks @charpeni preview looks as expected :)

File: docs/native-modules-android.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ Sometimes an app needs access to a platform API that React Native doesn't have a
 
 We designed React Native such that it is possible for you to write real native code and have access to the full power of the platform. This is a more advanced feature and we don't expect it to be part of the usual development process, however it is essential that it exists. If React Native doesn't support a native feature that you need, you should be able to build it yourself.
 
+
+## Native Module Setup
+
+Native modules are usually distributed as npm packages, except that for them to be native modules they will contain an Android library project. To get the basic scaffolding make sure to read [Native Modules Setup] guide first.
```
Comment (charpeni) [inline docs/native-modules-android.md]: Your link to `Native Modules Setup` doesn't work.

File: docs/native-modules-ios.md [hunk 2]
```diff
@@ -9,6 +9,10 @@ We designed React Native such that it is possible for you to write real native c
 
 This is a more advanced guide that shows how to build a native module. It assumes the reader knows Objective-C or Swift and core libraries (Foundation, UIKit).
 
+## Native Module Setup
+
+Native modules are usually distributed as npm packages, except that for them to be native modules they will contain an Xcode library project. To get the basic scaffolding make sure to read [Native Modules Setup] guide first.
```
Comment (charpeni) [inline docs/native-modules-ios.md]: Idem here on `Native Modules Setup` link.

**Comment by jugutier** [comment]
Hi @charpeni, sorry for the delay on this. I've rephrased that line and fixed the broken links. Let me know if it makes more sense now :)

**Comment by teameh** [comment]
Can't see why, but the page gives a 404.. https://facebook.github.io/react-native/docs/native-modules-setup.html

**Comment by jugutier** [comment]
I think this isn’t live yet

**Comment by endiliey** [comment]
Changes to docs/ are reflected in the next "master" version.

Try https://facebook.github.io/react-native/docs/next/native-modules-setup.html

**Comment by teameh** [comment]
I asked because the change on this page is already live and linking to it: https://facebook.github.io/react-native/docs/native-modules-ios#native-module-setup 

picture 2018-07-27 at 15 55 13


😄

**Comment by charpeni** [comment]
Thanks @teameh, it has been fixed on master.

**Pull request merged**